great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py

```diff
@@ -207,26 +243,80 @@ def get_metrics(
                 message=f"Utilizing a {self.__class__.__name__} requires a non-empty list of batch identifiers."
             )
 
-        domain_kwargs = build_metric_domain_kwargs(
+        """
+        Compute metrics, corresponding to multiple "MetricConfiguration" directives, together, rather than individually.
+
+        As a strategy, since "metric_domain_kwargs" changes depending on "batch_id", "metric_value_kwargs" serves as
+        identifying entity (through "AttributedResolvedMetrics") for accessing resolved metrics (computation results).
+
+        All "MetricConfiguration" directives are generated by combining each metric_value_kwargs" with
+        "metric_domain_kwargs" for all "batch_ids" (where every "metric_domain_kwargs" represents separate "batch_id").
+        Then, all "MetricConfiguration" objects, collected into list as container, are resolved simultaneously.
+        """
+
+        # Fist: Gather "metric_domain_kwargs" (corresponding to "batch_ids").
+
+        domain_kwargs: dict = build_metric_domain_kwargs(
             batch_id=None,
             metric_domain_kwargs=metric_domain_kwargs,
             domain=domain,
             variables=variables,
             parameters=parameters,
         )
 
-        metric_domain_kwargs: dict = copy.deepcopy(domain_kwargs)
+        batch_id: str
+        metric_domain_kwargs = [
+            copy.deepcopy(
+                build_metric_domain_kwargs(
+                    batch_id=batch_id,
+                    metric_domain_kwargs=copy.deepcopy(domain_kwargs),
+                    domain=domain,
+                    variables=variables,
+                    parameters=parameters,
+                )
+            )
+            for batch_id in batch_ids
+        ]
 
-        # Obtain value kwargs from "rule state" (i.e., variables and parameters); from instance variable otherwise.
-        metric_value_kwargs = get_parameter_value_and_validate_return_type(
-            domain=domain,
-            parameter_reference=metric_value_kwargs,
-            expected_return_type=None,
-            variables=variables,
-            parameters=parameters,
-        )
+        # Second: Gather "metric_value_kwargs" (caller may require same metric computed for multiple arguments).
+
+        if not isinstance(metric_value_kwargs, list):
+            metric_value_kwargs = [metric_value_kwargs]
+
+        value_kwargs_cursor: dict
+        metric_value_kwargs = [
+            # Obtain value kwargs from "rule state" (i.e., variables and parameters); from instance variable otherwise.
+            get_parameter_value_and_validate_return_type(
+                domain=domain,
+                parameter_reference=value_kwargs_cursor,
+                expected_return_type=None,
+                variables=variables,
+                parameters=parameters,
+            )
+            for value_kwargs_cursor in metric_value_kwargs
+        ]
```

Comment:
You were right about me not loving the comprehensions 😬 

I feel like a loop is more readable here but this is just nitpicking. Use what you think is best